tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js

```diff
@@ -326,16 +336,15 @@ module.exports = {
      *
      * @returns {Promise.<string[]>} Array of autocomplete webElementIds
      */
-    getShareAutocompleteWebElementIdList: function () {
+    getShareAutocompleteWebElementIdList: async function () {
       const webElementIdList = []
-      return this
-        .waitForElementVisible('@sharingAutoCompleteDropDownElements')
-        .api.elements('css selector', this.elements.sharingAutoCompleteDropDownElements.selector, (result) => {
-          result.value.forEach((value) => {
-            webElementIdList.push(value[Object.keys(value)[0]])
-          })
+      const autoCompleteDropdownSelector = this.elements.sharingAutoCompleteDropDownElements.selector
+      await this.api.elements('css selector', autoCompleteDropdownSelector, (result) => {
+        result.value.forEach((value) => {
+          webElementIdList.push(value[Object.keys(value)[0]])
         })
-        .then(() => webElementIdList)
+      })
+      return webElementIdList
```

Comment:
why do we have to change that implementation?
and what happened to the `.waitForElementVisible('@sharingAutoCompleteDropDownElements')`

Reply:
```..waitForElementVisible('@sharingAutoCompleteDropDownElements')``` checks for the visibility of auto-complete dropdown list. But in some cases, like if we disable sharing with groups or other cases, auto-complete dropdown may not be visible in the DOM and the test will fail. Therefore above code will check for sharingAutoCompleteDropDown and return an empty list if autocomplete dropdown is not visible.